src/selection.h

```diff
@@ -127,13 +138,23 @@ class Selection {
             {
                 if (cursor == cursorend) firstdx = dx;
                 int &curs = firstdx < 0 ? cursor : cursorend;
-                for (int c = curs, start = curs;;) {
+                int c = curs + dx;
+                wxChar ch;
+                if (c >= 0 && c <= MaxCursor()) {
+                    ch = GetCell()->text.t[min(c, curs)];
+                    //3,2,1 = whitespace, separator, anything else. If negative, only accepts same (abs) type
+                    int allowed = CharType(ch);
+                    curs = c;
+                    while(true) {
                     c += dx;
                     if (c < 0 || c > MaxCursor()) break;
-                    wxChar ch = GetCell()->text.t[(c + curs) / 2];
-                    if (!wxIsalnum(ch) && curs != start) break;
+                        ch = GetCell()->text.t[min(c,curs)];
+                        int chtype = CharType(ch);
+                        //disallow 3 decreases in type, i.e. "> #a" would become " #>a"

```

Comment:
The key concept is that there are three character types - whitespace, separator, and neither. Given these are presented with characters `w`/`s`/`n` respectively, the algorithm is able to skip `(w+s*)|(w+n*)|(s+n*)|(n+)` as represented by regex.
It does this by representing the types in an order (3/2/1) and setting the first type to be `allowed`. It allows a decrease in a type in a sequence (`chtype<=allowed`), but disallows a second decrease, which is represented by `allowed` being negative (`chtype != -allowed`, `if (chtype<allowed) allowed = -chtype;`). That is about all the algorithm does, I'm not sure how to make it even simpler, besides adding more comments to explain what it does.

Reply:
The reasoning behind doing at most a single decrease in type is because this allows skipping spaces after words, but needs at most one backtrack to get to either left or right border of any continuous sequence of `w`/`s`/`n`, whereas one would need two backtracks if it allowed `w+s*n*`.
VSCode actually has a different behaviour where it doesn't allow `s+n*` but only `sn*`, meaning if there is more than one separator it is treated as a separate sequence. However, this doesn't decrease the worst case amount of backtracking and makes the algorithm more complicated, so I left it out (plus it works just fine so far in my testing).

Reply:
Ok, well lets at least have the enum then, and some comments :)